index_websocket.js

```diff
@@ -0,0 +1,97 @@
+const http = require("http");
+const open = require("open");
+const fs = require("fs");
+const WebSocketServer = require("websocket").server; // websocket server class
+
+let webSocket = null;
+// connection variable, to refer to the web socket connection to the client once connection is established
+let connection = null;
+// default port used if none is specified by the user
+let defaultPort = 9101;
+const host = "http://127.0.0.1";
+
+// check if a browser window is already open
+let isRunning = false;
+
+
+exports.visualise = function visualise(jsonData, port = defaultPort) {
```

Comment:
For consistency with other languages which may require libraries for conversion to a JSON string, 
let's accept a JSON string as the argument to `visualise` for now i.e `visualise` won't call `JSON.stringify` internally.
```suggestion
exports.visualise = function visualise(jsonStr, port = defaultPort) {
```